components/brave_wallet/browser/zcash/zcash_orchard_storage.cc

```diff
@@ -0,0 +1,443 @@
+    std::copy(nullifier.begin(), nullifier.end(), nf.nullifier.begin());
+    result.push_back(std::move(nf));
+  }
+  if (!resolve_note_spents.Succeeded()) {
+    NOTREACHED_IN_MIGRATION();
+    return base::unexpected(Error{ErrorCode::kFailedToExecuteStatement,
+                                  database_.GetErrorMessage()});
+  }
+  return result;
+}
+
+base::expected<std::vector<OrchardNote>, ZCashOrchardStorage::Error>
+ZCashOrchardStorage::GetSpendableNotes(mojom::AccountIdPtr account_id) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  if (!EnsureDbInit()) {
+    NOTREACHED_IN_MIGRATION();
+    return base::unexpected(
+        Error{ErrorCode::kDbInitError, database_.GetErrorMessage()});
+  }
+
+  sql::Statement resolve_unspent_notes(
+      database_.GetUniqueStatement(base::StringPrintf(
+          "SELECT "
+          "notes.block_id, notes.amount,"
+          "notes.nullifier FROM %s LEFT OUTER JOIN spent_notes "
+          "ON notes.nullifier = spent_notes.nullifier AND notes.account_id = "
+          "spent_notes.account_id "
+          "WHERE spent_notes.nullifier IS NULL AND notes.account_id = ?;",
+          kNotesTable)));
+
+  resolve_unspent_notes.BindString(0, account_id->unique_key);
+
+  std::vector<OrchardNote> result;
+  while (resolve_unspent_notes.Step()) {
+    OrchardNote note;
+    auto block_id = ReadUint32(resolve_unspent_notes, 0);
+    auto amount = ReadUint32(resolve_unspent_notes, 1);
+    if (!block_id || !amount) {
+      NOTREACHED_IN_MIGRATION();
+      return base::unexpected(
+          Error{ErrorCode::kDbInitError, "Wrong database format"});
+    }
+    note.block_id = block_id.value();
+    note.amount = amount.value();
+    auto nullifier = resolve_unspent_notes.ColumnBlob(2);
+    std::copy(nullifier.begin(), nullifier.end(), note.nullifier.begin());
+    result.push_back(std::move(note));
+  }
+  return result;
+}
+
+std::optional<ZCashOrchardStorage::Error> ZCashOrchardStorage::UpdateNotes(
+    mojom::AccountIdPtr account_id,
+    const std::vector<OrchardNote>& found_notes,
+    const std::vector<OrchardNullifier>& spent_notes,
+    const uint32_t latest_scanned_block,
+    const std::string& latest_scanned_block_hash) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  CHECK(account_id);
+
+  if (!EnsureDbInit()) {
+    NOTREACHED_IN_MIGRATION();
+    return Error{ErrorCode::kDbInitError, database_.GetErrorMessage()};
+  }
+
+  sql::Transaction transaction(&database_);
+  if (!transaction.Begin()) {
+    NOTREACHED_IN_MIGRATION();
+    return Error{ErrorCode::kDbInitError, database_.GetErrorMessage()};
+  }
+
+  // Insert found notes to the notes table
+  sql::Statement statement_populate_notes(database_.GetUniqueStatement(
+      base::StringPrintf("INSERT INTO %s "
+                         "(account_id, amount, block_id, nullifier) "
+                         "VALUES (?, ?, ?, ?);",
+                         kNotesTable)));
+
```

Comment:
Everywhere it should be GetCachedStatement instead of GetUniqueStatement

Reply:
Done